mmcv/runner/hooks/logger/wandb.py

```diff
@@ -59,20 +60,26 @@ def before_run(self, runner):
             self.wandb.init()
         
         if self.config_path is not None:
-            shutil.copy2(self.config_path, self.wandb.run.dir)
             if os.path.isdir(self.config_path):
-                for path, _, _ in os.walk(self.wandb.run.dir):
+                copy_tree(self.config_path, self.wandb.run.dir)
+                for path_under_wandb, _, _ in os.walk(self.wandb.run.dir):
                     self.wandb.save(
-                        glob_str=path + '/*',
+                        glob_str=os.path.join(path_under_wandb,'/*'),
                         base_path=self.wandb.run.dir,
                         policy='now'
                     )
             else:
-                self.wandb.save(
-                    glob_str=self.wandb.run.dir + '/*',
-                    base_path=self.wandb.run.dir,
-                    policy='now'
-                )
+                if os.path.isfile(self.config_path):
+                    shutil.copy2(self.config_path, self.wandb.run.dir)
```

Comment:
Same as above

Reply:
This method was the best way in my experience. When I want to rebuild the same model with a specific config file, I sometimes experienced a hard time to find what I want to use because of a directory name. wandb `name` is a hashing value as default and not same as the `work-dir`. On the other hands, the copy version of config in `wandb.run.dir` could always have a high reliability.